src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateHistory.push({ ...addProperties(clone, action.extraData) });
+        break;
+
+      case 'removeProperties':
+        stateHistory.push({ ...removeProperties(clone, action.keysToRemove) });
```

Comment:
[CODE STYLE] - Similar to the 'addProperties' case, you don't need to spread the result of 'removeProperties' when pushing to 'stateHistory'.